lib/ddtrace/profiling/ext/thread.rb

```diff
@@ -0,0 +1,88 @@
+require 'ffi'
+
+module Datadog
+  module Profiling
+    module Ext
+      # C-struct for retrieving clock ID from pthread
+      class CClockId < FFI::Struct
+        layout :value, :int
+      end
+
+      # Extensions for pthread-backed Ruby threads, to retrieve
+      # the thread ID, clock ID, and CPU time.
+      module CThread
```

Comment:
This module is patching `Thread` itself, but do any of its functions need thread safety? (Mutexes etc?)

I don't think it does, given the fairly limited cases where state changes (PID/thread ID/clock ID only at thread start or process fork), and that adding mutexes might significantly impact performance, but worth being sure.

Reply:
The questions are:
1. is there any chance that multiple threads will execute any of the methods you added?
2. if yes, is it going to break or have inconsistent results?

For example, I don't think `initialize` can be called from multiple threads at the same time.
It sounds possible that you call `update_native_ids` concurrently, but in that case, nothing harmful would happen — you'd just do the job twice.